src/SDK/MonitoringApplianceEmulator/ViewModels/AlertDetailsControlViewModel.cs

```diff
@@ -143,5 +148,115 @@ private set
         public CommandHandler CloseControlCommand { get; }
 
         #endregion
+
+        private async Task<List<DisplayableAlertProperty>> ComposeArmProperties(AzureResourceManagerRequestAlertProperty armProperty, IAzureResourceManagerClient armClient)
+        {
+            var url = armProperty.AzureResourceManagerRequestUri;
+            IReadOnlyList<DisplayableAlertProperty> propertyRefs = armProperty.PropertiesToDisplay;
+            List<DisplayableAlertProperty> displayableArmProperties = new List<DisplayableAlertProperty>();
+            try
+            {
+                List<JObject> response = await armClient.ExecuteArmQueryAsync(url, CancellationToken.None);
+
+                foreach (IReferenceAlertProperty propertyRef in propertyRefs.OfType<IReferenceAlertProperty>())
+                {
+                    JToken propertyVal = response[0].SelectToken(propertyRef.ReferencePath);
```

Comment:
>JToken propertyVal = response[0].SelectToken(propertyRef.ReferencePath); [](start = 20, length = 72)

better check if the reference exists, and if not - add an "error property" just for this reference